src/export/StructureDefinitionExporter.ts

```diff
@@ -167,6 +168,12 @@ export class StructureDefinitionExporter implements Fishable {
                 rule.sourceInfo
               );
             } else {
+              if (!idRegex.test(invariant.name)) {
+                logger.error(
+                  `Invariant ${invariant.name} has a name which does not represent a valid FHIR id. FHIR ids may contain any combination of upper- or lower-case ASCII letters ('A'..'Z', and 'a'..'z'), numerals ('0'..'9'), '-' and '.', with a length limit of 64 characters.`,
+                  rule.sourceInfo
```

Comment:
You could potentially use the `InvalidFHIRIdError` here. I know we decided that the constraint should still be applied even if the Id is invalid, but you could throw the error after applying the constraint. If you don't want to use the error and want to keep this more specific message, that's fine, but I wanted to make sure you knew of it.